ToolBox/Polygon/PolygonDataQueueHandler.cs

```diff
@@ -507,105 +508,221 @@ private IEnumerable<Tick> GetCryptoTradeTicks(HistoryRequest request)
             }
         }
 
-
         private IEnumerable<Tick> GetEquityQuoteTicks(HistoryRequest request)
         {
             // https://api.polygon.io/v2/ticks/stocks/nbbo/SPY/2020-08-24?apiKey=
 
+            bool isException = false;
+
             var start = request.StartTimeUtc;
             var end = request.EndTimeUtc;
+            var offsetStart = Time.DateTimeToUnixTimeStampNanoseconds(start);
+            var offsetEnd = Time.DateTimeToUnixTimeStampNanoseconds(end);
 
-            while (start <= end)
+            // Continues until offsetStart gets to the end of the set time at offsetEnd
+            while (offsetStart < offsetEnd)
             {
+                // Skips Saturday and Sunday
+                if (!start.IsCommonBusinessDay())
+                {
+                    Log.Debug("Weekends are not processed " + start.ToString(System.Globalization.CultureInfo.GetCultureInfo("en-US")));
+                    start = start.Date.AddDays(1);
+                    break;
+                }
+                else
+                {
+                    Log.Debug("Processing " + start.ToString(System.Globalization.CultureInfo.GetCultureInfo("en-US")));
+                }
+
                 using (var client = new WebClient())
                 {
-                    var offset = Time.DateTimeToUnixTimeStampNanoseconds(start);
-                    var url = $"{HistoryBaseUrl}/v2/ticks/stocks/nbbo/{request.Symbol.Value}/{start.Date:yyyy-MM-dd}?apiKey={_apiKey}&timestamp={offset}";
+                    var url = $"{HistoryBaseUrl}/v2/ticks/stocks/nbbo/{request.Symbol.Value}/{start.Date:yyyy-MM-dd}?apiKey={_apiKey}&timestamp={offsetStart}&limit={_apiResultsLimit}";
 
-                    var response = client.DownloadString(url);
+                    Log.Debug(url);
```

Comment:
Let's make sure these logs follow the conversion of starting with the method name they are called from, for example: `Log.Debug(url);` -> `Log.Debug($"PolygonDataQueueHandler.GetEquityQuoteTicks(): target url {url}");`